development/extensions/tutorial.rst

```diff
@@ -0,0 +1,299 @@
+========
+Tutorial
+========
+
+Introduction
+============
+
+This tutorial explains the basic functionality of extensions:
+
+ * Basics: `Extension folder`_ and `composer.json`_
+ * `HTML Events`_
+ * `php Events`_
+ * Administration Module (ACP)
+ * Migrations (Database management)
+ * Controllers (Frontpage)
+ * Unit tests
+ * Automatic testing of changes on Travis CI
+
+.. seealso::
+
+   If you know modifications from phpBB 3.0, you may also have a look at
+   :doc:`modification_to_extension` which helps you to transform your
+   modification into an extension step by step.
+
+Extension folder
+================
+
+The first thing you need to do, in order to make phpBB aware of your extension,
+is to create a ``composer.json``. The file needs to be in a subfolder of the
+``ext/`` folder that is shipped with phpBB.
+Inside this folder you need to create a folder with your author (vendor) name.
+Then create the folder of the extension inside the vendor folder.
+
+.. note::
+
+    Both the vendor and extension folder names must start with a lower or
+    upper case ASCII letter (a to z), followed by ASCII letters and numbers
+    only!
+
+In this tutorial we use ``acme`` as vendor name and ``demo`` as extension name.
+So the ``composer.json`` should be located at ``ext/acme/demo/composer.json``.
+
+composer.json
+=============
+
+The content of the ``composer.json`` is very technical and contains the
```

Comment:
Well I added `is very technical and` to tell people not to worry too much, if they do not understand it.